lambdas/accesstoken/src/test/java/uk/gov/di/ipv/core/accesstoken/AccessTokenHandlerTest.java

```diff
@@ -166,6 +171,33 @@ void shouldReturn400OWhenInvalidAuthorisationCodeProvided() throws Exception {
         assertEquals(OAuth2Error.INVALID_GRANT.getDescription(), errorResponse.getDescription());
     }
 
+    @Test
+    void shouldReturn400WhenRedirectUrlInAuthGrantDoesNotMatchThatSuppliedToAuthEndpoint()
```

Comment:
Would it be worth adding a quick test to see what happens when `redirect_uri` isn't in the request? Not sure if it would cause a null pointer exception in `redirectUrlsDoNotMatch()`